class-duouniversal-wordpressplugin.php

```diff
@@ -181,7 +180,9 @@ function duo_authenticate_user( $user = '', $username = '', $password = '' ) {
 			$state = \sanitize_text_field( wp_unslash( $_GET['state'] ) );
 
 			// Retrieve the previously stored state and username from the session.
-			$associated_user = $this->get_username_from_oidc_state( $state );
+			$associated_user = $this->get_user_from_oidc_state( $state );
+			// Cleanup state
+			\delete_site_option( "duo_auth_state_$state" );
```

Comment:
I do like that idea, although I couldn't really come up with a refactoring beyond changing the method to something like `get_and_delete_user_from_oidc_state`, which like you said doesn't buy us much since this is this only usage.

Also, in the happy path this deletion isn't actually even necessary here, as all of the duo auth values are erased at the start of `duo_start_second_factor`, so a lingering previous state _should_ be erased. I added this delete just as an extra safeguard to prevent orphan buildup in any unhappy-path scenarios.